ipynb/agp_distance_decay.ipynb

```diff
@@ -0,0 +1,279 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This Jupyter notebook recreates the distance-decay analyses for the American Gut Project paper by McDonald et al.\n",
+    "\n",
+    "Ashkaan K Fahimipour\n",
+    "2016.01.21"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "## Let's first install and load some libraries.\n",
+    "install.packages('biom', repos = \"http://cran.rstudio.com/\")\n",
+    "install.packages('phyloseq', repos = \"http://cran.rstudio.com/\")\n",
+    "install.packages('qiimer', repos = \"http://cran.rstudio.com/\")\n",
+    "install.packages('ggplot2', repos = \"http://cran.rstudio.com/\")\n",
+    "install.packages('vegan', repos = \"http://cran.rstudio.com/\")\n",
+    "install.packages('fields', repos = \"http://cran.rstudio.com/\")\n",
+    "install.packages('dplyr', repos = \"http://cran.rstudio.com/\")\n",
+    "library(biom)\n",
+    "library(phyloseq)\n",
+    "library(qiimer)\n",
+    "library(ggplot2)\n",
+    "library(vegan)\n",
+    "library(fields)\n",
+    "library(dplyr)\n",
+    "\n",
+    "## I was given a 1-column csv file with sample IDs to include in my analyses\n",
+    "use <- read.csv('filepath/file.csv', header = TRUE)\n",
+    "use[, 1] <- as.character(use[, 1])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we'll read in the BIOM table into R for analysis. R really hates non-JSON formatted BIOM tables (e.g., tables generated with macqiime v1.9), so you may have to use BIOM-convert in to reformat to JSON before importing:\n",
+    "\n",
+    "http://biom-format.org/documentation/biom_conversion.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "## import the table\n",
+    "biom <- read_biom('filepath/file.biom')\n",
+    "\n",
+    "## subset data by 'Samples To Use' file\n",
+    "dat <- biom_data(biom)\n",
+    "dat <- as.data.frame(as(biom_data(biom), \"matrix\"), header = TRUE)\n",
+    "dat <- dat[, which(names(dat) %in% use[, 1])]\n",
+    "\n",
+    "## subset again by samples that only occur in >= 10 samples per Josh Ladau's instructions\n",
+    "dat <- dat[-c(which(rowSums(dat != 0) < 10)), ]\n",
+    "\n",
+    "## pull out metadata\n",
+    "tax <- as(observation_metadata(biom), \"matrix\")\n",
+    "meta <- as(sample_metadata(biom), \"matrix\")\n",
+    "meta <- as.data.frame(samp)\n",
+    "\n",
```

Comment:
@AshkaanF, would it be possible to include the translations necessary to operate directly from the publicly available summarized data (ftp://ftp.microbio.me/AmericanGut/)?